.github/workflows/main.yml

```diff
@@ -361,6 +361,53 @@ jobs:
           name: 'obs-studio-windows-${{ matrix.arch }}-${{ steps.setup.outputs.commitHash }}'
           path: '${{ env.FILE_NAME }}'
 
+  source_build:
+    name: '02 - Source'
+    runs-on: [ubuntu-22.04]
+    if: always()
+    needs: [clang_check]
+    defaults:
+      run:
+        shell: bash
+        working-directory: 'obs-studio'
+    steps:
+      - name: 'Checkout'
+        uses: actions/checkout@v3
+        with:
+          submodules: 'recursive'
+          path: 'obs-studio'
+          fetch-depth: 0
+
+      - name: 'Setup environment'
+        id: setup
+        run: |
+          echo "commitHash=$(git rev-parse --short=9 HEAD)" >> $GITHUB_OUTPUT
+
+      - name: 'Create source artifact'
+        if: ${{ success() && github.event_name != 'pull_request' }}
+        run: |
+          prefix=obs-studio-${{ steps.setup.outputs.commitHash }}
+          commit_date=$(git show -s --format=%cI ${{ steps.setup.outputs.commitHash }})
+          outfile=${prefix}.tar.gz
+          ARTIFACT_NAME=${outfile}
+          outfile=${{ github.workspace }}/${outfile}
+          echo "outfile: $outfile"
+          echo "artifact: $ARTIFACT_NAME"
+          # create a tarball with *all* the sources
+          # - prefix paths with 'obs-studio/' (since we are creating the tarball from within the source dir)
+          # - exclude all repository files
+          # - make the tarball reproducible
+          tar -cvzf ${outfile}  --transform='s|^\.|obs-studio|'  --exclude-vcs --exclude-vcs-ignores --exclude '.git*'  --sort=name  --owner=0 --group=0 --numeric-owner --pax-option="exthdr.name=%d/PaxHeaders/%f,delete=atime,delete=ctime" --mtime="${commit_date}"  .
+          # announce artifact for upload
+          echo "FILE_NAME=${ARTIFACT_NAME}" >> $GITHUB_ENV
```

Comment:
This has no need to be an env var, turn it into an output.

Reply:
i've (rebased on current master and) force pushed a version that now uses outputs rather than envvars.

#### run only on tag/release

afaict, the current pipelines will generate the following artifacts on each push:
|artifact|size|
|--------|----|
|obs-studio-flatpak-5b370d4e3-x86_64|139 MB|
|obs-studio-macos-arm64-5b370d4e3|146 MB|
|obs-studio-macos-x86_64-5b370d4e3|156 MB|
|obs-studio-ubuntu-20.04-5b370d4e3|108 MB|
|obs-studio-ubuntu-20.04-5b370d4e3-dbgsym|80.8 MB|
|obs-studio-ubuntu-22.04-5b370d4e3|108 MB|
|obs-studio-ubuntu-22.04-5b370d4e3-dbgsym|72.9 MB|
|obs-studio-windows-x64-5b370d4e3|186 MB|
|obs-studio-windows-x86-5b370d4e3|167 MB|
|**TOTAL**|**1160MB**|

the source-job adds another artifact with a size of approximately 32MB.
The worry that this job imposes a problem, seems a bit ... exaggerated to me.

But anyhow: I see that e.g. the `Windows Installer` resp `macOS notarized image` jobs seem to only run for releases by using

```yaml
if: ${{ startsWith(github.ref, 'refs/tags/') && github.event_name != 'pull_request' }}
```

if you want I can add this condition (but that means that the pipeline won't be run before you merge, which imho is even less desirable; but ymmv, so just tell me if you require me to do so).